app/scripts/services/paymentReportService.js

```diff
@@ -0,0 +1,95 @@
+angular
+  .module('confRegistrationWebApp')
+  .service('paymentReportService', function PaymentReportService(
+    $cacheFactory,
+    $rootScope,
+    $http,
+    $q,
+  ) {
+    const path = function(id) {
+      return 'conferences/' + (id || '') + '/payments/report';
+    };
+
+    function getReportData(url) {
+      const defer = $q.defer();
```

Comment:
Can we not use `defer`? You should just be able to return the promise returned by `$http.get` directly.

Reply:
fixed